dissect/target/plugins/os/unix/linux/_os.py

```diff
@@ -33,15 +33,16 @@ def detect(cls, target: Target) -> Optional[Filesystem]:
         return None
 
     @export(property=True)
-    def ips(self) -> list[str]:
+    @arg("--dhcp-all", action="store_true", help="parse all syslog, messages and journal files for DHCP IP addresses")
+    def ips(self, dhcp_all: bool = False) -> list[str]:
```

Comment:
I think this is inherit to properties, so no arguments can be passed to it. This might be incorrectly be reflected in the documentation. 

Also, I remembered a similar discussion in #133 where we talked about a similar change where we decided to move it to `ips_dhcp` in the meantime. It might be a good idea to move towards this again, to avoid the `t.ip` vs `t.ips()` inconsistency across OSes.

Reply:
I'll raise this discussion in the team and get back to you!

Reply:
We had a brief discussion and decided that's not an ideal solution. You can execute multiple functions, and each function can have it's own argument parser. There's no "one big" set of parsed arguments available at runtime, each plugin only gets to see its own set of parsed argument that matched its defined arguments.

For something like properties I think it'd be better to use dependency injection, but that's non-trivial to add and we rarely use properties anyway, so I fear it may be a bit of a waste to add support for that.

For this specific case we expect this to be resolved by the network interface rework that's been in the planning for a while, described in #773. 

An alternative to `ips_dhcp` to work around this issue in the meantime is to (ab)use the target configuration system. At the moment, this can be used by placing a file called `.targetcfg.py` in the same directory (or parents) of a target, and storing a variable like `UNIX_IPS_DHCP = True` in there. You can access this through `target._config.UNIX_IPS_DHCP` in the code. Note that this behaves like a module type so you may need to `getattr`. We're open to improvements to this config system by the way, e.g. adding support for environment variables (#771, #772).

Or revert back to `ips_dhcp` :smile: but both of these will be temporary workarounds.